modules/Core/pages/panel/integrations.php

```
@@ -56,19 +56,25 @@
         $errors = [];
 
         if (Token::check()) {
-            DB::getInstance()->update('integrations', $integration->data()->id, [
-                'enabled' => Output::getClean(Input::get('enabled')),
-                'can_unlink' => Output::getClean(Input::get('can_unlink')),
-                'required' => Output::getClean(Input::get('required'))
-            ]);
-
-            Session::flash('integrations_success', $language->get('admin', 'integration_updated_successfully'));
-            Redirect::to(URL::build('/panel/core/integrations/', 'integration=' . $integration->getName()));
+            if (Input::get('action') === 'general_settings') {
+                DB::getInstance()->update('integrations', $integration->data()->id, [
+                    'enabled' => Output::getClean(Input::get('enabled')),
+                    'can_unlink' => Output::getClean(Input::get('can_unlink')),
+                    'required' => Output::getClean(Input::get('required'))
+                ]);
+
+                Session::flash('integrations_success', $language->get('admin', 'integration_updated_successfully'));
+                Redirect::to(URL::build('/panel/core/integrations/', 'integration=' . $integration->getName()));
+            }
         } else {
             $errors[] = $language->get('general', 'invalid_token');
         }
     }
 
+    if ($integration->getSettings() !== null && file_exists($integration->getSettings())) {
```

Wouldn't it make more sense to throw an error if the integration settings file doesn't exist? Rather than just silently ignoring it. I feel like this can cause some frustration for developers that forget it.  And the settings file should never be configured anyways if the file doesn't exist. This can result in unwanted behavior in case a file was not uploaded correctly

---

Done